### ISSUE TITLE
`@stylistic/selector-descendant-combinator-no-non-space` passes over a selector carrying an inline comment

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#158]
Found while fixing [#124](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/124), which taught the rule to read a selector carrying a block comment. A selector carrying an inline comment is a different story, and is passed over for now.

### What minimal example or steps are needed to reproduce the bug?

```scss
.foo  // comment
  .bar {}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "customSyntax": "postcss-scss",
  "rules": {
    "@stylistic/selector-descendant-combinator-no-non-space": true
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.scss" --config myconfig.json --fix
```

### What did you expect to happen?

The two spaces in front of the comment are reported at the position they stand at and collapsed, the comment keeping its `//` spelling.

### What actually happened?

Nothing is reported. The rule passes over any selector whose raws carry a `scss` field.

### Do you have a proposal to fix the bug?

`postcss-scss` rewrites every inline comment of a selector into a block comment inside `raws.selector.raw`, keeps the source spelling in `raws.selector.scss`, and prints the latter — `scss-stringifier.js` reads `raw.scss ? raw.scss : raw.raw`. Two things follow, and both have to be answered at once:

- the two strings drift apart by two characters per inline comment, so an index measured against the parsed selector points past where the problem stands in the source;
- a fixed selector written into `raws.selector.raw` never reaches the output, and Stylelint counts a fixer as applied whatever it does, so `--fix` would end with a clean pass on a file it had not touched.

`selectorCombinatorSpaceChecker` already answers both, for `selector-combinator-space-before` and `selector-combinator-space-after`: it pairs every comment node with the text it has in the source, converts indices back with the accumulated delta, and writes the block spelling into `raw` and the inline one into `scss`. That machinery is private to the util and would have to be shared before this rule could use it.

There is one thing beyond it, though. A run of whitespace standing *after* an inline comment cannot be collapsed at all: such a comment ends only with a line break, so writing a single space in its place would comment the rest of the selector out. `.foo // c\n  .bar` therefore has a problem that can only be reported, never fixed — while the run standing *in front of* the comment collapses safely.
